parse return value of remove_if

```
cards.cpp: In member function ‘void Cards::organize()’:
cards.cpp:121:27: error: ignoring return value of ‘_FIter std::remove_if(_FIter, _FIter, _Predicate) [with _FIter = __gnu_cxx::__normal_iterator<_SkillSpec<unsigned int>*, vector<_SkillSpec<unsigned int> > >; _Predicate = Cards::organize()::<lambda(const SkillSpec&)>]’, declared with attribute ‘nodiscard’ [-Werror=unused-result]
  121 |             std::remove_if(card->m_skills.begin(), card->m_skills.end(), [](const SkillSpec& ss) {return ss.id == Skill::summon;});
      |             ~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /usr/lib/gcc/x86_64-pc-linux-gnu/15/include/g++-v15/algorithm:63,
                 from /usr/include/boost/token_functions.hpp:38,
                 from /usr/include/boost/token_iterator.hpp:24,
                 from /usr/include/boost/tokenizer.hpp:20,
                 from cards.cpp:3:
/usr/lib/gcc/x86_64-pc-linux-gnu/15/include/g++-v15/bits/stl_algo.h:806:5: note: declared here
  806 |     remove_if(_ForwardIterator __first, _ForwardIterator __last,
      |     ^~~~~~~~~
````

--- a/cards.cpp
+++ b/cards.cpp
@@ -118,7 +118,10 @@ void Cards::organize()
         {
             std::cerr << "WARNING: Card [" << card->m_id << "] (" << card->m_name
                 << ") summons an unknown card [" << summon_card_id << "] (removing invalid skill Summon)" << std::endl;
-            std::remove_if(card->m_skills.begin(), card->m_skills.end(), [](const SkillSpec& ss) {return ss.id == Skill::summon;});
+            card->m_skills.erase(
+                std::remove_if(card->m_skills.begin(), card->m_skills.end(), [](const SkillSpec& ss) {return ss.id == Skill::summon;}),
+                card->m_skills.end()
+            );
             card->m_skill_value[Skill::summon] = 0;
         }
     }

